File: AutomatizareDemoQA.java

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.apache.poi.ss.usermodel.*;
import org.apache.poi.xssf.usermodel.XSSFWorkbook;

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;

        public class AutomatizareDemoQA extends Principal{
        public static void main(String[] args) {
            Principal ob = new Principal();
            ob.beforeTest();
            driver.get("https://demoqa.com/webtables");

            FileInputStream fis = null;
            try {
                fis = new FileInputStream(new File("demoqa.xlsx"));
                Workbook wb = new XSSFWorkbook(fis);
                Sheet sheet = wb.getSheetAt(0);
                for (Row row : sheet) {
                    if (row.getRowNum() == 0) continue; // skip the header row

                    String nume = row.getCell(0).getStringCellValue();
                    String prenume = row.getCell(1).getStringCellValue();
                    int varsta = (int) row.getCell(2).getNumericCellValue();
                    String email = row.getCell(3).getStringCellValue();
                    int salariu = (int) row.getCell(4).getNumericCellValue();
                    String profesie = row.getCell(5).getStringCellValue();

                    WebElement btn_add = driver.findElement(By.id("addNewRecordButton"));
                    btn_add.click();

                    WebElement first_name_input = driver.findElement(By.id("firstName"));
                    WebElement last_name_input = driver.findElement(By.id("lastName"));
                    WebElement email_input = driver.findElement(By.id("userEmail"));
                    WebElement age_input = driver.findElement(By.id("age"));
                    WebElement salary_input = driver.findElement(By.id("salary"));
                    WebElement departament_input = driver.findElement(By.id("department"));
                    WebElement btn_submit = driver.findElement(By.id("submit"));

                    first_name_input.sendKeys(nume);
                    last_name_input.sendKeys(prenume);
                    email_input.sendKeys(email);
                    age_input.sendKeys(String.valueOf(varsta));
                    salary_input.sendKeys(String.valueOf(salariu));
                    departament_input.sendKeys(profesie);
                    btn_submit.click();
                }
            } catch (Exception e) {
                System.out.println("Virsta de " + e.getMessage() + " nu a putut fi introdusa");
            } finally {
                System.out.println("Exceptia a luat sfirsit");
                try {
                    if (fis != null) fis.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
                driver.quit();
            }
        }
    }

File: AutomatizareDemoQA.py
from selenium import webdriver
from selenium.webdriver.common.by import By
import openpyxl

#setam browserul firefox si dimensiunile la maxim
driver = webdriver.Firefox()
driver.maximize_window()
# accesam pagina
driver.get('https://demoqa.com/webtables')

# deschidem fisierul excel si foaia de lucru din care vom extrage date
wb = openpyxl.load_workbook("demoqa.xlsx")
sheet = wb.active

# asteptam 3 secunde inainte de a incepe testul
driver.implicitly_wait(3)
try:
    # iteram prin fiecare celula din excel si introducem valorile in variabilele de mai jos
    for row in sheet.iter_rows(min_row=2, values_only=True):
        nume, prenume, varsta, email, salariu, profesie = row
        # identificam campurile de care avem nevoie
        btn_add = driver.find_element(By.ID, 'addNewRecordButton')
        btn_add.click()
        first_name_input = driver.find_element(By.ID, "firstName")
        last_name_input = driver.find_element(By.ID, 'lastName')
        email_input = driver.find_element(By.ID, 'userEmail')
        age_input = driver.find_element(By.ID, 'age')
        salary_input = driver.find_element(By.ID, 'salary')
        departament_input = driver.find_element(By.ID, 'department')
        btn_submit = driver.find_element(By.ID, 'submit')

    # completam campurile cu valorile necesare
        first_name_input.send_keys(nume)
        last_name_input.send_keys(prenume)
        email_input.send_keys(email)
        age_input.send_keys(str(varsta))
        salary_input.send_keys(str(salariu))
        departament_input.send_keys(profesie)
        btn_submit.click()
except:
    print(f"Virsta de {age_input} nu a putut fi introdusa")
finally:
    print("Exceptia a luat sfirsit")

    # asteptam 2 secunde si apoi inchidem browserul
driver.implicitly_wait(2)
driver.quit()
